Backwards-compatibility patch for defaultTemperature fix

greentfrapp opened the pull request:
Maps defaultTemperature to F

gcampax commented on nlp/compat.js:
```diff
@@ -120,10 +120,22 @@ function deviceNames(locale, result, entities) {
 
+function defaultUnits(locale, result, entitites) {
+    for (let i = 0; i < result.code.length; i++) {
+        if (result.code[i].startsWith('default')) {
```
This will run in NN syntax, so the token will be `unit:defaultTemperature`, and it should be replaced with `unit:F`.

greentfrapp replied:
Fixed!

gcampax commented:
You should probably add tests to tests/unit/test_compat.js

greentfrapp commented:
Added tests to test_nlp_compat. @gcampax I also fixed what I think was a bug in `test_nlp_compat.js` that caused only first 4 tests to run. Can you confirm that it was a bug and not intentional?
